src/gui/uithememanager.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2019  Prince Gupta <jagannatharjun11@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If
+ * you modify file(s), you may extend this exception to your version of the
+ * file(s), but you are not obligated to do so. If you do not wish to do so,
+ * delete this exception statement from your version.
+ */
+
+#include "uithememanager.h"
+
+#include <QApplication>
+#include <QFile>
+#include <QResource>
+
+#include "base/logger.h"
+#include "base/preferences.h"
+
+UIThemeManager *UIThemeManager::m_instance = nullptr;
+
+void UIThemeManager::freeInstance()
+{
+    if (m_instance) {
+        delete m_instance;
+        m_instance = nullptr;
+    }
+}
+
+void UIThemeManager::initInstance()
+{
+    if (!m_instance)
+        m_instance = new UIThemeManager;
+}
+
+UIThemeManager::UIThemeManager()
+{
+    if (Preferences::instance()->useCustomUITheme()) {
+        QResource::registerResource(Preferences::instance()->customUIThemePath()
+                                    , "/uitheme");
+    }
+}
+
+UIThemeManager *UIThemeManager::instance()
+{
+    return m_instance;
+}
+
+void UIThemeManager::applyStyleSheet()
+{
+    if (Preferences::instance()->useCustomUITheme()) {
+        QFile qssFile(":uitheme/stylesheet.qss");
+        if (!qssFile.open(QIODevice::ReadOnly | QIODevice::Text))
+            LogMsg("Can't open :/uitheme/stylesheet.qss, error: "
+                   + qssFile.errorString()
+                   , Log::WARNING);
+        qApp->setStyleSheet(qssFile.readAll()); // empty stylesheet on error
+    }
+    else
```

Comment:
Now it make sense (to me) to return early: `if (!Preferences::instance()->useCustomUITheme()) {/*setStyleSheet && return*/}`
because you can save an indention in a long block.

Reply:
#1
```
void UIThemeManager::applyStyleSheet()
{
    QFile qssFile(":uitheme/stylesheet.qss");
    if (!Preferences::instance()->useCustomUITheme() || !qssFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
        qApp->setStyleSheet({}); // use default stylesheet
        return;
    }
    qApp->setStyleSheet(qssFile.readAll());
}
```
#2
```
void UIThemeManager::applyStyleSheet()
{
    if (!Preferences::instance()->useCustomUITheme()) {
        qApp->setStyleSheet({});
        return;
    }
    QFile qssFile(":uitheme/stylesheet.qss");
    if (qssFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
        qApp->setStyleSheet(qssFile.readAll());
        return;
    }
    LogMsg(QString(tr("Can't open :/uitheme/stylesheet.qss, error: %1")).arg(qssFile.errorString())
           , Log::WARNING);
    qApp->setStyleSheet({});
}
```

which one?

Reply:
neither
```c++
void UIThemeManager::applyStyleSheet()
{
    if (!Preferences::instance()->useCustomUITheme()) {
        qApp->setStyleSheet({});
        return;
    }

    QFile qssFile(":uitheme/stylesheet.qss");
    if (!qssFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
        LogMsg(QString(tr("Can't open :/uitheme/stylesheet.qss, error: %1")).arg(qssFile.errorString())
            , Log::WARNING);
        qApp->setStyleSheet({});
        return;
    }

    qApp->setStyleSheet(qssFile.readAll());
}
```